src/pages/signup.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react"
+import { useForm } from "react-hook-form"
+import "../styles/signup.css"
+
+export default function App({ location }) {
+  const [isSubmitted, setIsSubmitted] = useState(false)
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm()
+  const onSubmit = data => {
+    setIsSubmitted('spinner')
+    fetch(`/api/signup-submit`, {
+      method: `POST`,
+      body: JSON.stringify(data),
+      headers: {
+        "content-type": `application/json`,
+      },
+    })
+      .then(res => res.json())
+      .then(body => {
+        console.log(`response from API:`, body)
+        setIsSubmitted(body)
+      }).catch(err => {
+        console.error(`Error updating calendar:`, err)
+        setIsSubmitted(`Error updating calendar: ${err}`)
+      })
+  }
+
+  const params = new URLSearchParams(location.search)
+
+  if (isSubmitted == 'spinner') {
+    return (
+      <div className="signup form-container">
+        <h2>Please wait</h2>
+        <p>Updating the calendar and sending invite...</p>
+      </div>
+    )
+
+  } else if (isSubmitted) {
+    if (isSubmitted == 'success') {
+      return (
+        <div className="signup form-container">
+          <h2>Thank you for signing up!</h2>
+          <p>Please check your inbox for a calendar invite for <b>{params.get("title")}</b>.</p>
+        </div>
+      )
+    } else {
+      return (
+        <div className="signup form-container"><pre>{isSubmitted}</pre></div>
+      )
+    }
+  } else {
```

Comment:
This nested else if else is hard to read. Perhaps move the markup to a functional component or method that is then composed within the if else logic?